### PR TITLE
Fixes to RMG tools

### LINKED ISSUE #118: PopulateReactions not working properly
The website was not updated following ReactionMechanismGenerator/RMG-Py@bfa7516fc8b80691d414c76e18099bece4d1cde1, so the PopulateReactions tool does not work at all, as can be seen on the dev site.

However, even after making the appropriate change (5ca3188632e51125e2713d9829b5e7e152dfda61), the tool still does not seem to work properly when tested locally. Specifically, it will not find a single reaction. Testing the RMG-Py script directly from the command line gives the expected result, so the problem appears to be on the website side. The live website is currently working, but when I checked out the same commits locally, it also gave zero reactions.

One difference I noticed is that the website will give the following warning

```
Should only make one instance of RMGDatabase because it's stored as a module-level variable!
Unexpected behaviour may result!
```

while running from the command line will not, which I assume is because the website preloads the database? However, I don't know why that would result in not finding any reactions, and I'm also extremely confused as to how the live website is working.

I'm also working on writing unit tests for this.

### PR DESCRIPTION
**Pull request by mliu49**
- Addresses some issues I came across while testing RMG tools.
- Added some unittests for RMG tools.
    - Done using Django client to simulate POST request and check response
    - Primarily tests to make sure that the pages aren't broken and expected outputs are created
    - Whether the links on the webpage correctly link to the output files is not being tested, but really should be
- Added new page for getting current Git commits for RMG-Py, RMG-database, and RMG-website following last week's discussion
    - No links to the page have been added yet: need to decide where to put the link and how visible it is should be
    - Page can be accessed by going directly to /version
- Revert to using subprocess call for Populate Reactions to avoid conflicts with the global RMGDatabase object - Fixes #118

### CODE REVIEW

**Comment by mliu49** [comment]
I think this is ready to be merged, pending any further comments or suggestions.

**Comment by nickvandewiele** [comment]
Great work, @mliu49 : With these unit tests, an automated Travis CI build could be triggered now!

**Comment by mliu49** [comment]
The issue with SimpleTestCase is due to changes in [Django 1.9](https://docs.djangoproject.com/en/1.9/releases/1.9/#miscellaneous) (I've been working with Django 1.8.4).

**Comment by connie** [comment]
Should we change the environment.yml to use 1.8.4? Rather than 1.9.0.  Or should we try to upgrade to 1.9.0?

**Comment by mliu49** [comment]
We probably want to move to 1.9 eventually, but we can specify 1.8.4 for now, until we're fully compatible. I can work on 1.9 compatibility later on.

I think the current error is also due to changes in Django 1.9 with regard to [file uploading](https://docs.djangoproject.com/en/1.9/releases/1.9/#file-storage). It seems that Django tries to assign more unique filenames to uploaded files?

**Comment by rwest** [comment]
@hamza-khan could you comment on upgrading to Django 1.9? (Although perhaps that discussion deserves a separate issue number)

**Comment by connie** [comment]
Sounds good.  Let's try 1.8.4 first and get this working before migrating.

File: environment.yml [hunk 1]
```diff
@@ -30,7 +30,7 @@ dependencies:
   - symmetry
   - mopac
   - graphviz
-  - django
+  - django ==1.84
```
Comment (mliu49) [inline environment.yml]: I think you missed a period.

**Comment by connie** [comment]
Sigh.. still failing.

**Comment by mliu49** [comment]
Yeah, I'm not sure what the problem is now, since it works locally for me. I just freshly cloned onto my mac, essentially following the travis build steps, and everything passed.

**Comment by mliu49** [comment]
Finally figure out what the problem was. We were specifying the full path for uploading files, while we should actually only specify the path relative to MEDIA_ROOT. I assume some change happened somewhere between django 1.3 and 1.8, but for some reason didn't error when run locally.

I'm going to clean up this branch, and also refactor file uploading to be more straightforward. This might end up making the changes in the migrations branch unnecessary.

**Comment by mliu49** [comment]
This should pass after ReactionMechanismGenerator/RMG-Py#734 is merged.

**Comment by connie** [comment]
Works great!

**Pull request merged**